### PR TITLE
Py3 compat, take II

### PR DESCRIPTION
**Pull request by blizzz**
Rebase of #146, keep fingers crossed, if successful all kudos to @ethifus

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -29,14 +29,16 @@
...
         "requests >= 2.0.1",
+        "six"
     ],
-	classifiers=[
-		'Programming Language :: Python',
-		'Development Status :: 3 - Alpha',
-		'Environment :: Web Environment',
+    classifiers=[
+        'Programming Language :: Python',
+        'Programming Language :: Python :: 3.4',
```
Comment (PVince81) [inline setup.py]: I wonder whether we need to provide two versions for pip now, one for Python 2 and one for Python 3

**Comment by PVince81** [comment]
Looks good, great work ! :+1: 

My only concern is about packaging, need to find out how other projects that support both Python 2 and Python 3 do it.

**Comment by blizzz** [comment]
I have no insights there unfortunately. However, pyhn has a package  that is 2 and 3 compatible: https://pypi.python.org/pypi/pyhn/0.3.7, apparently it boils down to specify the supported python versions by a classifier (cf. http://www.diveintopython3.net/packaging.html, here a list https://pypi.python.org/pypi?:action=list_classifiers), i.e. adjusting setup.py. Currently it just says "Python", so it shows up with pip3:

```
$ pip3 search pyocclient
pyocclient (0.2)  - Python client library for ownCloud
```

I suggest adding ``Programming Language :: Python :: 2`` and ``Programming Language :: Python :: 3`` with this pull request. Then it should work as usual (i have no idea how to push this to pypi, but I guess you figured it out). Do you want me to add these classifiers?

**Comment by PVince81** [comment]
@blizzz sounds good, please add the classifiers :smile:

**Comment by SergioBertolinSG** [comment]
Tests running with python3 fails:
```
 File "owncloud/test/test.py", line 5, in <module>
    from unittest_data_provider import data_provider
ImportError: No module named 'unittest_data_provider'
```

**Comment by SergioBertolinSG** [comment]
Tests running with python3 fails:

```
 File "owncloud/test/test.py", line 5, in <module>
    from unittest_data_provider import data_provider
ImportError: No module named 'unittest_data_provider'
```

unittest module is not available in pip3, I have installed unittest2.

**Comment by blizzz** [comment]
@SergioBertolinSG works for me:

```
$ pip3 search unittest | grep data
…
unittest-data-provider (1.0.1)     - PHPUnit-like @dataprovider decorator for unittest
…
```

**Comment by SergioBertolinSG** [comment]
@blizzz yes that works, thanks.

Tried with both python2 and python3, there is one failure but not related with this changes.

There is a warning .owncloud/test/test.py:472: DeprecationWarning: Please use assertEqual instead.

:+1:

**Comment by blizzz** [comment]
> @blizzz sounds good, please add the classifiers :smile: 

done :snake:

**Comment by PVince81** [comment]
Weird, now the unit tests have failures

File: setup.py [hunk 2]
```diff
@@ -33,7 +33,8 @@
     ],
     classifiers=[
         'Programming Language :: Python',
-        'Programming Language :: Python :: 3.4',
+        'Programming Language :: Python :: 2'
+        'Programming Language :: Python :: 3'
```
Comment (PVince81) [inline setup.py]: you missed the commas

Comment (blizzz) [inline setup.py]: :scream:

**Comment by blizzz** [comment]
Now, still failing. This is strange, isn't it.

**Comment by PVince81** [comment]
Yeah, strange. Maybe Travis updated their Python version or changed the warning flags ?

The error seems about a file that did not get closed properly. Maybe we're missing a fclose statement somewhere ?

**Comment by blizzz** [comment]
> Yeah, strange. Maybe Travis updated their Python version or changed the warning flags ?

The last ones passing where also 2.7.10 and 3.5.0. Maybe some module got an update.

> The error seems about a file that did not get closed properly. Maybe we're missing a fclose statement somewhere ?

Maybe on 3. On 2.7 it's just ``InvalidHeader: Header value 1 must be of type str or bytes, not <type 'int'>`` deep in the requests module.

**Comment by SergioBertolinSG** [comment]
Yes the error happens after the upgrade to requests-2.11.0, I tested this with requests-2.10.0 and it worked fine.

**Comment by blizzz** [comment]
Woohoo :+1: @SergioBertolinSG

**Comment by PVince81** [comment]
@blizzz wanna add this thing to the changelog with credits yourself ? Else I'll do it later at release time.

**Comment by blizzz** [comment]
the credits should go to @ethifus really

**Comment by PVince81** [comment]
Ok, added https://github.com/owncloud/pyocclient/commit/7af7a5aef06a55a9b33015593896a47ac0c9e753

**Comment by blizzz** [comment]
thx 🎩

**Pull request merged**